### PR TITLE
Clarify MariaDB

### PR DESCRIPTION
**Pull request by coltoneshaw**
#### Summary
Clarifying where MariaDB stands due to the mysql 5.7.12 requirement. Maria no longer acts as a drop in replacement and customers are experiencing issues when they upgrade because of this.

Maria docs - https://mariadb.com/kb/en/mariadb-vs-mysql-compatibility/

### CODE REVIEW

**Comment by coltoneshaw** [comment]
Probably can reword this to be more clear, but was struggling. :)

File: source/install/software-hardware-requirements.rst [hunk 1]
```diff
@@ -114,6 +114,8 @@ For Mattermost v5.39 and previous releases:
 
 Amazon Aurora equivalents of both MySQL and PostgreSQL are also supported.
 
+Maria v10+ no longer functions as a MySQL drop-in replacement which means it is not supported for Mattermost due to the requirement of MySQL 5.7.12. Prior versions were not officially supported but may have functioned in older Mattermost releases. If you are running MariaDB now, migrating from Maria to the MySQL equivalent is suggested.
```
Comment (cwarnermm) [inline source/install/software-hardware-requirements.rst:117]:
```suggestion
MariaDB v10+ no longer functions as a MySQL drop-in replacement, and it's not supported for Mattermost due to the requirement of MySQL 5.7.12. Prior versions of MariaDB were not officially supported but may have functioned in older Mattermost releases. If you are running MariaDB now, migrating from MariaDB to the MySQL equivalent is suggested.
```

Comment (cwarnermm) [reply source/install/software-hardware-requirements.rst:117]: Suggested or recommended?

**Review by cwarnermm — COMMENTED** [review]
Thanks so much, @coltoneshaw, for this important docs update! I've provided feedback inline.

File: source/install/software-hardware-requirements.rst (hunk 1 above)
Comment (coltoneshaw) [reply source/install/software-hardware-requirements.rst:117]: I mean, it's the **only** real option they have. I was just trying to be nice.

Comment (cwarnermm) [reply source/install/software-hardware-requirements.rst:117]: Updated the verbiage to recommended. :)

**Review by svelle — APPROVED** [review]
LGTM

**Review by cwarnermm — APPROVED** [review]
